### DISCUSSION TITLE
Josh Goldberg: TypeScript ESLint, New Book

### DISCUSSION THREAD

**Discussion by hipstersmoothie** [devtools-fm/devtools.fm#89] [Announcements]
## Background:

[Twitter](https://twitter.com/joshuakgoldberg)
[Website](https://www.joshuakgoldberg.com/)
[Book](https://www.oreilly.com/library/view/learning-typescript/9781098110321/)
[TypeScript ESLint](https://github.com/typescript-eslint/typescript-eslint)
- type system gymnastics
  - https://www.youtube.com/watch?v=7lyb22x9tcM
  - https://www.youtube.com/watch?v=7nj3vgZjL0Y

## Intro

Andrew: Hello, welcome to the devtools.fm podcast. This is a podcast about developer tools and the people who make them. I'm Andrew and this is my co-host Justin. 

Justin: Hey everyone! Our guest today is Josh Goldberg. Josh is an author, speaker, and open source maintainer in the TypeScript ecosystem. He recently published a new book called _Learning TypeScript_. Congrats on that Josh! Is there anything else you'd like to tell our audience about yourself?

## Questions

- [x] What originally got you into typescript?
- [x] I know you primarily for your talks at tsconf. Your approach to learning the type system in such a unique way is entertaining. Could you recount some of the ways you've abused the type system for fun?
- [x] You're a maintainer on typescript eslint, how did you get involved in the project? What is the project and why is it useful?
- [x] Have you written any interesting rules?
- [x] Have you built any other typescript based tooling?
- [x] Congrats on the book release. What did you just release? Who's the intended audience?
- [x] What topics do you cover?
- [x] What part of the book are you most proud of?
- [x] Does the book come with any hands on projects?
- [x] What are some examples of the included projects?
- [x] Any plans to write another book?
- [x] What yet to be implemented typescript feature are you dying to have?
- [x] What's a recent typescript feature you love?

## Tooltips

Add them in the comments below!

## Outro

Andrew: Well that’s it for this week’s episode of devtools.fm. Be sure to follow us on YouTube and wherever you consume your podcasts. Thanks for listening!

**Comment by JoshuaKGoldberg** [comment]
This looks great! 

> Type system gymnastics

I'm publishing a blog post ~today~ soon about _"Extreme Explorations in the TypeScript Type System"_ that covers a bunch of similar things other folks have made too. https://github.com/LearningTypeScript/site/pull/55 -> it'll be on https://www.learningtypescript.com/articles/extreme-explorations-of-typescripts-type-system ~later today~ likely midday EST Monday.

> Tooltips

* https://github.com/mattpocock/ts-error-translator (https://ts-error-translator.vercel.app) by [@mattpocock](https://twitter.com/mattpocockuk). Great explainer for TypeScript errors.
* _[Laziness Does Not Exist](https://www.simonandschuster.com/books/Laziness-Does-Not-Exist/Devon-Price/9781797120591)_ by Devon Price, Ph.D. Helping me feel better about my work ethics & avoid burnout.

**Comment by hipstersmoothie** [comment]
My Tooltips:

https://www.npmjs.com/package/jscodeshift
https://react-cmdk.com

**Comment by just-be-dev** [comment]
- https://interval.com/
- https://github.com/coddingtonbear/obsidian-local-rest-api
